docs/usage/session.rst

```diff
@@ -0,0 +1,59 @@
+.. _session:
+
+==================
+Session management
+==================
+
+Zyte API provides powerful session APIs:
+
+-   Client-managed sessions give you full control over session management.
+
+-   :ref:`Server-managed sessions <zyte-api-session-contexts>` let Zyte API
+    handle session management for you.
+
+When using scrapy-zyte-api, you can use these session APIs through the
+corresponding Zyte API fields (:http:`request:session`,
+:http:`request:sessionContext`).
+
+However, scrapy-zyte-api also provides :ref:`its own session API
+<plugin-sessions>`.
+
+.. _plugin-sessions:
+
+scrapy-zyte-api’s session API
+=============================
+
+scrapy-zyte-api’s session API offers a higher-level API on top of
+client-managed sessions, to enjoy some of the advantages of client-managed
+sessions (e.g. expiring specific sessions) while removing some of their
+drawbacks (management overhead).
+
+To use scrapy-zyte-api’s session API, define
+:setting:`ZYTE_API_SESSION_PARAMS` or :setting:`ZYTE_API_SESSION_CHECKER`.
+
+Often it makes sense to define both. For example:
+
+.. code-block:: python
+    :caption: settings.py
+
+    ZYTE_API_SESSION_PARAMS = {
```

Comment:
Can it be part of `ZYTE_API_AUTOMAP_PARAMS` setting?

Reply:
Also, the middleware is designed to also work with requests using manual parameters (`zyte_api` / `ZYTE_API_DEFAULT_PARAMS`) and scrapy-poet integration (`zyte_api_provider` / `ZYTE_API_PROVIDER_PARAMS`).

Reply:
> Then they would be used both when initializing a session and when using the session with a regular request. Do we really want that?

To elaborate:

At the moment, if you define:

```python
ZYTE_API_SESSION_PARAMS = {"actions": [{"action": "setLocation", {"address": {"postalCode": "04662"}}}], "geolocation": "GB"}
ZYTE_API_AUTOMAP_PARAMS = {"geolocation": "GB"}
```

When sending a request, a session is created with `ZYTE_API_SESSION_PARAMS`, and then the actual request is sent with `ZYTE_API_AUTOMAP_PARAMS` and `session` set.

If instead you had:

```python
ZYTE_API_AUTOMAP_PARAMS = {"actions": [{"action": "setLocation", {"address": {"postalCode": "04662"}}}], "geolocation": "GB"}
```

And used `ZYTE_API_AUTOMAP_PARAMS` for request initialization, the request initialization and the actual request would both be using the `setLocation` action, which would be pointless and prevent you from using an HTTP request after session initialization.

If what you are suggesting is that given:

```python
ZYTE_API_SESSION_PARAMS = {"actions": [{"action": "setLocation", {"address": {"postalCode": "04662"}}}]}
ZYTE_API_AUTOMAP_PARAMS = {"geolocation": "GB"}
```

Then session initialization should include the geolocation from `ZYTE_API_AUTOMAP_PARAMS`, that could become problematic, because some other parameters from `ZYTE_API_AUTOMAP_PARAMS` might not be compatible with the session initialization parameters (i.e. with browser requests), e.g.

```python
ZYTE_API_SESSION_PARAMS = {"actions": [{"action": "setLocation", {"address": {"postalCode": "04662"}}}]}
ZYTE_API_AUTOMAP_PARAMS = {"geolocation": "GB", "device": "mobile"}
```

This would cause a 400 response from Zyte API during session initialization.

Reply:
> Is sessios not going to be initialized by the ZAPI's script?

To initialize sessions with a custom Zyte API script, you would define:

```python
ZYTE_API_SESSION_PARAMS = {"actions": [{"action": "interaction", "id": "<SCRIPT_ID>"}]}
```

If you mean Zyte API will automatically run some session initialization script when you create a client-managed session, then you could skip this parameter altogether.